src/base/associative_cache.hh

```diff
@@ -0,0 +1,347 @@
+/*
+ * Copyright (c) 2024 Pranith Kumar
+ * All rights reserved
+ *
+ * The license below extends only to copyright in the software and shall
+ * not be construed as granting a license to any other intellectual
+ * property including but not limited to intellectual property relating
+ * to a hardware implementation of the functionality of the software
+ * licensed hereunder.  You may use the software subject to the license
+ * terms below provided that you ensure that this notice is replicated
+ * unmodified and in its entirety in all distributions of the software,
+ * modified or unmodified, in source code or in binary form.
```

Comment:
If possible, it would be better to remove this "extra" clause and use the BSD license in the LICENSE file in the root directory.

Reply:
Done.